Lombiq.Hosting.Tenants.Maintenance/Maintenance/RemoveLoginInfos/RemoveLoginInfosMaintenanceProvider.cs

```diff
@@ -0,0 +1,44 @@
+using Lombiq.Hosting.Tenants.Maintenance.Extensions;
+using Lombiq.Hosting.Tenants.Maintenance.Models;
+using Lombiq.Hosting.Tenants.Maintenance.Services;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Options;
+using OrchardCore.Users;
+using OrchardCore.Users.Models;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace Lombiq.Hosting.Tenants.Maintenance.Maintenance.RemoveLoginInfos;
+
+public class RemoveLoginInfosMaintenanceProvider : MaintenanceProviderBase
+{
+    private readonly IOptions<RemoveLoginInfosMaintenanceOptions> _options;
+    private readonly ISession _session;
+    private readonly UserManager<IUser> _userManager;
+
+    public RemoveLoginInfosMaintenanceProvider(
+        IOptions<RemoveLoginInfosMaintenanceOptions> options,
+        ISession session,
+        UserManager<IUser> userManager)
+    {
+        _options = options;
+        _session = session;
+        _userManager = userManager;
+    }
+
+    public override Task<bool> ShouldExecuteAsync(MaintenanceTaskExecutionContext context) =>
+        Task.FromResult(
+            _options.Value.IsEnabled &&
+            !context.WasLatestExecutionSuccessful());
+
+    public override async Task ExecuteAsync(MaintenanceTaskExecutionContext context)
+    {
+        var users = await _session.Query<User>().ListAsync();
+        foreach (var user in users)
+        {
+            user.LoginInfos.RemoveAll();
+            await _userManager.UpdateAsync(user);
```

Comment:
After testing it this way:

1. In appsettings set IsEnabled to false.
2. app_data delete
3. autosetup
4. Login with Microsoft Account
5. Stop the app
6. In appsettings set IsEnabled to true.
7. Azure AD login should be deleted after this.
8. In user menu in external logins Azure AD should not be present.

When I try to login with Azure AD, I got this: 
![image](https://github.com/Lombiq/Hosting-Tenants/assets/80963259/6190254b-7025-46a8-92a4-2ae7b1850f26)

I tried it with my previous password but it didn't work. I guess it should behave as the first time.

Reply:
I implemented a new approach now.

Reply:
It works now.